mm2src/coins/z_coin.rs

```diff
@@ -734,6 +737,32 @@ pub async fn z_coin_from_conf_and_params(
     builder.build().await
 }
 
+fn verify_checksum_zcash_params(
+    spend_path: PathBuf,
+    expected_spend_hash: H256,
+    output_path: PathBuf,
+    expected_out_hash: H256,
+) -> Result<bool, ZCoinBuildError> {
+    let spend_bytes = std::fs::read(spend_path)?;
+    let out_bytes = std::fs::read(output_path)?;
```

Comment:
These files are quite big (`sapling-spend` is almost 50Mb), so reading them into memory entirely is quite an overhead. Please consider using the approach like https://stackoverflow.com/a/71606608.

Reply:
done, works as expected